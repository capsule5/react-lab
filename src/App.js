import React from "react"
import { Route, Redirect } from "react-router-dom"
import Examples from "./components/core/examples/Examples"
import "./App.css"


const App = () => (
  <main className="App">
    <Route path="/" component={ Examples } />
    <Redirect to="/examples/list" />
  </main>
)

export default App

