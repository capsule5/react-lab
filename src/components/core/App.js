import React from "react"
import { Route, Redirect } from "react-router-dom"
import Examples from "./examples/Examples"


const App = () => (
  <main className="App">
    <Route path="/" component={ Examples } />
    <Redirect to="/examples/list" />
  </main>
)

export default App

