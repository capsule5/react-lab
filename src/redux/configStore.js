import { createStore, combineReducers, applyMiddleware } from "redux"
import thunkMiddleware from "redux-thunk"
import { createLogger } from "redux-logger"
// reducers
import potagers from "./potagers/potagersReducer"


const middlewares = [ thunkMiddleware ]

if (process.env.NODE_ENV !== "production") {
  const loggerMiddleware = createLogger()
  middlewares.push(loggerMiddleware)
}

const rootReducer = combineReducers({
  potagers,
})

export default function configureStore(initialState) {
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(
      ...middlewares
    )
  )

  return store
}
