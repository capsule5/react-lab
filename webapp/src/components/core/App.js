import React from "react"
import { Route, Redirect, Switch } from "react-router-dom"
import Examples from "./examples/Examples"


const App = () => (
  <main className="App">
    <Switch>
      <Route path="/" component={ Examples } />
      <Redirect to="/examples/list" />
    </Switch>
  </main>
)

export default App

