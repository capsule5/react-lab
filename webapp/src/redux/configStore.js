import { createStore, combineReducers, applyMiddleware } from "redux"
import thunkMiddleware from "redux-thunk"
import { createLogger } from "redux-logger"
// reducers
import { tagsReducer as tags } from "./tags"


const middlewares = [ thunkMiddleware ]

if (process.env.NODE_ENV !== "production") {
  const loggerMiddleware = createLogger()
  middlewares.push(loggerMiddleware)
}

const rootReducer = combineReducers({
  tags,
})

export default function configureStore(initialState) {
  const store = createStore(
    rootReducer,
    initialState,
    applyMiddleware(
      ...middlewares
    )
  )

  return store
}
