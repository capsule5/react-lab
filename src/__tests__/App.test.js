import React from "react"
import { BrowserRouter } from "react-router-dom"
import { Provider } from "react-redux"
import configStore from "redux/configStore"
import App from "App"
import renderer from "react-test-renderer"

const store = configStore()

it("renders without crashing", () => {
  const component = renderer.create(
    <Provider store={ store }>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
