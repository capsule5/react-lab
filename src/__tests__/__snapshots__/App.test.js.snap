// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<main
  className="App"
>
  <div
    className="Layout"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        JT React Lab
      </h1>
    </header>
    <div
      className="sc-iwsKbI gIVEvE"
    >
      <div
        className="sc-jTzLTM fprOSV"
      >
        <nav
          className="sc-gqjmRU ghNqmd"
        >
          <ul>
            <li
              className="list"
            >
              ≡ 
              <a
                href="/examples/list"
                onClick={[Function]}
              >
                List view (
                7
                )
              </a>
            </li>
            <span>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  › 
                  <a
                    href="/examples/animation-list"
                    onClick={[Function]}
                  >
                    Animation List
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  › 
                  <a
                    href="/examples/transitiongroup-gsap"
                    onClick={[Function]}
                  >
                    TransitionGroup / GSAP
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  › 
                  <a
                    href="/examples/transitiongroup-animejs"
                    onClick={[Function]}
                  >
                    TransitionGroup / Anime.js
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  › 
                  <a
                    href="/examples/unmount-animation"
                    onClick={[Function]}
                  >
                    Unmount Animation
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  › 
                  <a
                    href="/examples/unmount-animation-with-rm"
                    onClick={[Function]}
                  >
                    Unmount Animation with RM
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  › 
                  <a
                    href="/examples/simple-transition"
                    onClick={[Function]}
                  >
                    Simple Transition
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  › 
                  <a
                    href="/examples/animated-value"
                    onClick={[Function]}
                  >
                    Animated Value
                  </a>
                </li>
              </div>
            </span>
          </ul>
        </nav>
        <div
          className="content"
        >
          <div>
            <span>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <div
                  className="sc-bxivhb diTrAk"
                >
                  <h1>
                    Animation List
                  </h1>
                  <div
                    className="sc-ifAKCX gxohuR"
                  >
                    <div
                      className="data sc-bwzfXH kajHfT"
                    >
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        animation
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        react-transition-group
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        gsap
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        unmounting
                      </a>
                    </div>
                    
                    <ul
                      className="data sc-htpNat jkiVGe"
                    >
                      <li
                        className="false false "
                      >
                        <a
                          href="https://greensock.com/gsap"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://greensock.com/gsap
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="content"
                  >
                    <span>
                      <div
                        className="sc-dnqmqq eMVKko"
                      >
                        pilote
                      </div>
                      <div
                        className="sc-dnqmqq eMVKko"
                      >
                        industrie
                      </div>
                      <div
                        className="sc-dnqmqq eMVKko"
                      >
                        de
                      </div>
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <div
                  className="sc-bxivhb diTrAk"
                >
                  <h1>
                    TransitionGroup / GSAP
                  </h1>
                  <div
                    className="sc-ifAKCX gxohuR"
                  >
                    <div
                      className="data sc-bwzfXH kajHfT"
                    >
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        animation
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        react-transition-group
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        gsap
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        unmounting
                      </a>
                    </div>
                    
                    <ul
                      className="data sc-htpNat jkiVGe"
                    >
                      <li
                        className="false false "
                      >
                        <a
                          href="https://www.freshtilledsoil.com/whats-the-most-developer-friendly-react-animation-library/"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://www.freshtilledsoil.com/whats-the-most-developer-friendly-react-animation-library/
                        </a>
                        <div
                          className="expendBtn"
                          onClick={[Function]}
                        >
                          ...
                        </div>
                      </li>
                      <li
                        className="isExpendable false "
                      >
                        <a
                          href="https://reactjs.org/docs/animation.html"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://reactjs.org/docs/animation.html
                        </a>
                      </li>
                      <li
                        className="isExpendable false "
                      >
                        <a
                          href="https://greensock.com/gsap"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://greensock.com/gsap
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="content"
                  >
                    <div>
                      <div>
                        <button
                          hidden={false}
                          onClick={[Function]}
                        >
                          enter animation
                        </button>
                        <button
                          hidden={true}
                          onClick={[Function]}
                        >
                          exit animation
                        </button>
                      </div>
                      <div>
                        <span />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <div
                  className="sc-bxivhb diTrAk"
                >
                  <h1>
                    TransitionGroup / Anime.js
                  </h1>
                  <div
                    className="sc-ifAKCX gxohuR"
                  >
                    <div
                      className="data sc-bwzfXH kajHfT"
                    >
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        animation
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        react-transition-group
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        anime.js
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        unmounting
                      </a>
                    </div>
                    
                    <ul
                      className="data sc-htpNat jkiVGe"
                    >
                      <li
                        className="false false "
                      >
                        <a
                          href="https://www.freshtilledsoil.com/whats-the-most-developer-friendly-react-animation-library/"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://www.freshtilledsoil.com/whats-the-most-developer-friendly-react-animation-library/
                        </a>
                        <div
                          className="expendBtn"
                          onClick={[Function]}
                        >
                          ...
                        </div>
                      </li>
                      <li
                        className="isExpendable false "
                      >
                        <a
                          href="https://reactjs.org/docs/animation.html"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://reactjs.org/docs/animation.html
                        </a>
                      </li>
                      <li
                        className="isExpendable false "
                      >
                        <a
                          href="http://animejs.com/"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          http://animejs.com/
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="content"
                  >
                    <div>
                      <div>
                        <button
                          hidden={false}
                          onClick={[Function]}
                        >
                          enter animation
                        </button>
                        <button
                          hidden={true}
                          onClick={[Function]}
                        >
                          exit animation
                        </button>
                      </div>
                      <div>
                        <span />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <div
                  className="sc-bxivhb diTrAk"
                >
                  <h1>
                    Unmount Animation
                  </h1>
                  <div
                    className="sc-ifAKCX gxohuR"
                  >
                    <div
                      className="data sc-bwzfXH kajHfT"
                    >
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        animation
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        unmounting
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        transition-events
                      </a>
                    </div>
                    <div
                      className="data"
                    >
                      Based on transition events
                    </div>
                    <ul
                      className="data sc-htpNat jkiVGe"
                    >
                      <li
                        className="false false "
                      >
                        <a
                          href="https://stackoverflow.com/questions/40064249/react-animate-mount-and-unmount-of-a-single-component"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://stackoverflow.com/questions/40064249/react-animate-mount-and-unmount-of-a-single-component
                        </a>
                        <div
                          className="expendBtn"
                          onClick={[Function]}
                        >
                          ...
                        </div>
                      </li>
                      <li
                        className="isExpendable false "
                      >
                        <a
                          href="https://reactjs.org/docs/events.html#transition-events"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://reactjs.org/docs/events.html#transition-events
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="content"
                  >
                    <button
                      onClick={[Function]}
                    >
                      Unmount
                    </button>
                    <div
                      className="sc-bZQynM dobSaP"
                      onTransitionEnd={[Function]}
                    >
                      Unmount me!
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <div
                  className="sc-bxivhb diTrAk"
                >
                  <h1>
                    Unmount Animation with RM
                  </h1>
                  <div
                    className="sc-ifAKCX gxohuR"
                  >
                    <div
                      className="data sc-bwzfXH kajHfT"
                    >
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        animation
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        react-motion
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        unmounting
                      </a>
                    </div>
                    
                    <ul
                      className="data sc-htpNat jkiVGe"
                    />
                  </div>
                  <div
                    className="content"
                  >
                    <button
                      onClick={[Function]}
                    >
                      Unmount
                    </button>
                    <div>
                      <div
                        className="sc-htoDjs gFKPzo"
                        style={
                          Object {
                            "opacity": 1,
                          }
                        }
                      >
                         Unmount me! 
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <div
                  className="sc-bxivhb diTrAk"
                >
                  <h1>
                    Simple Transition
                  </h1>
                  <div
                    className="sc-ifAKCX gxohuR"
                  >
                    <div
                      className="data sc-bwzfXH kajHfT"
                    >
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        animation
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        react-motion
                      </a>
                    </div>
                    
                    <ul
                      className="data sc-htpNat jkiVGe"
                    >
                      <li
                        className="false false "
                      >
                        <a
                          href="https://github.com/chenglou/react-motion/blob/master/demos/demo0-simple-transition/index.html"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://github.com/chenglou/react-motion/blob/master/demos/demo0-simple-transition/index.html
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="content"
                  >
                    <button
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      Toggle
                    </button>
                    <div
                      className="sc-gzVnrw cIcbfg"
                    >
                      <div
                        className="demo0"
                      >
                        <div
                          className="demo0-block"
                          style={
                            Object {
                              "WebkitTransform": "translate3d(0px, 0, 0)",
                              "transform": "translate3d(0px, 0, 0)",
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <div
                  className="sc-bxivhb diTrAk"
                >
                  <h1>
                    Animated Value
                  </h1>
                  <div
                    className="sc-ifAKCX gxohuR"
                  >
                    <div
                      className="data sc-bwzfXH kajHfT"
                    >
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        animation
                      </a>
                      <a
                        className="light false sc-bdVaJa emDXLY"
                        href=""
                        onClick={[Function]}
                      >
                        #
                        react-motion
                      </a>
                    </div>
                    
                    <ul
                      className="data sc-htpNat jkiVGe"
                    >
                      <li
                        className="false false "
                      >
                        <a
                          href="https://github.com/chenglou/react-motion"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://github.com/chenglou/react-motion
                        </a>
                        <div
                          className="expendBtn"
                          onClick={[Function]}
                        >
                          ...
                        </div>
                      </li>
                      <li
                        className="isExpendable false "
                      >
                        <a
                          href="https://react.rocks/tag/react-motion"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          https://react.rocks/tag/react-motion
                        </a>
                      </li>
                    </ul>
                  </div>
                  <div
                    className="content"
                  >
                    <div
                      className="sc-EHOje enfYfd"
                    >
                      0
                    </div>
                  </div>
                </div>
              </div>
            </span>
          </div>
        </div>
        <nav
          className="sc-VigVT bJnPWW"
        >
          <ul>
            <span>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  <a
                    className="inSideNav false sc-bdVaJa emDXLY"
                    href=""
                    onClick={[Function]}
                  >
                    #
                    animation
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  <a
                    className="inSideNav false sc-bdVaJa emDXLY"
                    href=""
                    onClick={[Function]}
                  >
                    #
                    anime.js
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  <a
                    className="inSideNav false sc-bdVaJa emDXLY"
                    href=""
                    onClick={[Function]}
                  >
                    #
                    gsap
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  <a
                    className="inSideNav false sc-bdVaJa emDXLY"
                    href=""
                    onClick={[Function]}
                  >
                    #
                    react-motion
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  <a
                    className="inSideNav false sc-bdVaJa emDXLY"
                    href=""
                    onClick={[Function]}
                  >
                    #
                    react-transition-group
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  <a
                    className="inSideNav false sc-bdVaJa emDXLY"
                    href=""
                    onClick={[Function]}
                  >
                    #
                    transition-events
                  </a>
                </li>
              </div>
              <div
                className="sc-gZMcBi gBCTZz"
              >
                <li>
                  <a
                    className="inSideNav false sc-bdVaJa emDXLY"
                    href=""
                    onClick={[Function]}
                  >
                    #
                    unmounting
                  </a>
                </li>
              </div>
            </span>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</main>
`;
